Assets/MixedRealityToolkit/UX/Scripts/BoundingBoxes/BoundingBoxGizmoHandle.cs

```diff
@@ -0,0 +1,312 @@
+        }
+        private void ApplyRotation(Vector3 currentHandPosition)
+        {
+            Vector3 initialRay = initialHandPosition - transformToAffect.position;
+            initialRay.Normalize();
+
+            Vector3 currentRay = currentHandPosition - transformToAffect.position;
+            currentRay.Normalize();
+
+            Vector3 delta = currentRay - initialRay;
+            delta.Scale(rotationFromPositionScale);
+
+            Vector3 newEulers = new Vector3(0, 0, 0);
+            if (Axis == AxisToAffect.X)
+            {
+                newEulers = new Vector3(-delta.y, 0, 0);
+            }
+            else if (Axis == AxisToAffect.Y)
+            {
+                newEulers = new Vector3(0, delta.x, 0);
+            }
+            else if (Axis == AxisToAffect.Z)
+            {
+                newEulers = new Vector3(0, 0, delta.y);
+            }
+
+            if (IsLeftHandedRotation)
+            {
+                newEulers.Scale(new Vector3(-1.0f, -1.0f, -1.0f));
+            }
+
+            newEulers += initialOrientation;
+
+            transformToAffect.rotation = Quaternion.Euler(newEulers);
+        }
+
+        public void OnInputDown(InputEventData eventData)
+        {
+            inputDownEventData = eventData;
+
+            initialHandPosition     = GetHandPosition(eventData.SourceId);
+            initialScale            = transformToAffect.localScale;
+            initialPosition         = transformToAffect.position;
+            initialOrientation      = transformToAffect.rotation.eulerAngles;
+            initialHandOrientation  = GetHandOrientation(eventData.SourceId);
+            initialScaleOrigin      = transformToAffect.position - this.transform.position;
+
+            MixedRealityToolkit.InputModule.InputManager.Instance.PushModalInputHandler(gameObject);
+
+            this.gameObject.GetComponent<Renderer>().material = Rig.InteractingMaterial;
+            Rig.FocusOnHandle(this.gameObject);
+            eventData.Use();
+        }
+        public void OnInputUp(InputEventData eventData)
+        {
+            inputDownEventData = null;
+
+            if (this.AffineType == TransformType.Scale)
+            {
+                this.gameObject.GetComponent<Renderer>().material = Rig.ScaleHandleMaterial;
+            }
+            else
+            {
+                this.gameObject.GetComponent<Renderer>().material = Rig.RotateHandleMaterial;
+            }
+
+            MixedRealityToolkit.InputModule.InputManager.Instance.PopModalInputHandler();
+            Rig.FocusOnHandle(null);
+
+            if (eventData != null)
+            {
+                eventData.Use();
+            }
+        }
+        public void OnSourceDetected(SourceStateEventData eventData)
+        {
+        }
+        public void OnSourceLost(SourceStateEventData eventData)
+        {
+            if (eventData.SourceId == inputDownEventData.SourceId)
+            {
+                OnInputUp(null);
```

Comment:
We probably shouldn't be calling this method like that.
Might be better if we had a private method that handed the logic